// ionic-angular/components/segment/segment.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/linker/element';
import * as import3 from 'ionic-angular/components/segment/segment';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from '@angular/core/src/linker/element_ref';
import * as import9 from '@angular/core/src/metadata/view';
import * as import10 from '@angular/core/src/linker/component_factory';
var renderType_SegmentButton_Host = null;
var _View_SegmentButton_Host0 = (function (_super) {
    __extends(_View_SegmentButton_Host0, _super);
    function _View_SegmentButton_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_SegmentButton_Host0, renderType_SegmentButton_Host, import6.ViewType.HOST, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_SegmentButton_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('ion-segment-button', rootSelector, null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'segment-button');
        this.renderer.setElementAttribute(this._el_0, 'role', 'button');
        this.renderer.setElementAttribute(this._el_0, 'tappable', '');
        this._appEl_0 = new import2.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_SegmentButton0(this.viewUtils, this.injector(0), this._appEl_0);
        this._SegmentButton_0_4 = new import3.SegmentButton(this.renderer, new import8.ElementRef(this._el_0));
        this._appEl_0.initComponent(this._SegmentButton_0_4, [], compView_0);
        compView_0.create(this._SegmentButton_0_4, this.projectableNodes, null);
        var disposable_0 = this.renderer.listen(this._el_0, 'click', this.eventHandler(this._handle_click_0_0.bind(this)));
        this.init([].concat([this._el_0]), [this._el_0], [disposable_0], []);
        return this._appEl_0;
    };
    _View_SegmentButton_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import3.SegmentButton) && (0 === requestNodeIndex))) {
            return this._SegmentButton_0_4;
        }
        return notFoundResult;
    };
    _View_SegmentButton_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        if (((this.numberOfChecks === 0) && !throwOnChange)) {
            this._SegmentButton_0_4.ngOnInit();
        }
        this.detectContentChildrenChanges(throwOnChange);
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_SegmentButton_Host0.prototype._handle_click_0_0 = function ($event) {
        this._appEl_0.componentView.markPathToRootAsCheckOnce();
        var pd_0 = (this._SegmentButton_0_4.onClick() !== false);
        return (true && pd_0);
    };
    return _View_SegmentButton_Host0;
}(import1.AppView));
function viewFactory_SegmentButton_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_SegmentButton_Host === null)) {
        (renderType_SegmentButton_Host = viewUtils.createRenderComponentType('', 0, import9.ViewEncapsulation.None, [], {}));
    }
    return new _View_SegmentButton_Host0(viewUtils, parentInjector, declarationEl);
}
export var SegmentButtonNgFactory = new import10.ComponentFactory('ion-segment-button', viewFactory_SegmentButton_Host0, import3.SegmentButton);
var styles_SegmentButton = [];
var renderType_SegmentButton = null;
var _View_SegmentButton0 = (function (_super) {
    __extends(_View_SegmentButton0, _super);
    function _View_SegmentButton0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_SegmentButton0, renderType_SegmentButton, import6.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_SegmentButton0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this.renderer.projectNodes(parentRenderNode, import4.flattenNestedViewRenderNodes(this.projectableNodes[0]));
        this._el_0 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'button-effect');
        this.init([], [this._el_0], [], []);
        return null;
    };
    return _View_SegmentButton0;
}(import1.AppView));
export function viewFactory_SegmentButton0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_SegmentButton === null)) {
        (renderType_SegmentButton = viewUtils.createRenderComponentType('/home/nicolas/projects/cloudtripper/cloudtripper-frontend/node_modules/ionic-angular/components/segment/segment.d.ts class SegmentButton - inline template', 1, import9.ViewEncapsulation.None, styles_SegmentButton, {}));
    }
    return new _View_SegmentButton0(viewUtils, parentInjector, declarationEl);
}
