// ionic-angular/components/fab/fab.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/linker/element';
import * as import3 from 'ionic-angular/components/fab/fab';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from 'ionic-angular/config/config';
import * as import9 from '@angular/core/src/linker/element_ref';
import * as import10 from '@angular/core/src/metadata/view';
import * as import11 from '@angular/core/src/linker/component_factory';
import * as import12 from 'ionic-angular/components/icon/icon';
import * as import13 from '@angular/core/src/linker/query_list';
var renderType_FabButton_Host = null;
var _View_FabButton_Host0 = (function (_super) {
    __extends(_View_FabButton_Host0, _super);
    function _View_FabButton_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_FabButton_Host0, renderType_FabButton_Host, import6.ViewType.HOST, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_FabButton_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('div', rootSelector, null);
        this.renderer.setElementAttribute(this._el_0, 'ion-fab', '');
        this._appEl_0 = new import2.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_FabButton0(this.viewUtils, this.injector(0), this._appEl_0);
        this._FabButton_0_4 = new import3.FabButton(this.parentInjector.get(import8.Config), new import9.ElementRef(this._el_0), this.renderer);
        this._appEl_0.initComponent(this._FabButton_0_4, [], compView_0);
        compView_0.create(this._FabButton_0_4, this.projectableNodes, null);
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_FabButton_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import3.FabButton) && (0 === requestNodeIndex))) {
            return this._FabButton_0_4;
        }
        return notFoundResult;
    };
    return _View_FabButton_Host0;
}(import1.AppView));
function viewFactory_FabButton_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_FabButton_Host === null)) {
        (renderType_FabButton_Host = viewUtils.createRenderComponentType('', 0, import10.ViewEncapsulation.None, [], {}));
    }
    return new _View_FabButton_Host0(viewUtils, parentInjector, declarationEl);
}
export var FabButtonNgFactory = new import11.ComponentFactory('[ion-fab]', viewFactory_FabButton_Host0, import3.FabButton);
var styles_FabButton = [];
var renderType_FabButton = null;
var _View_FabButton0 = (function (_super) {
    __extends(_View_FabButton0, _super);
    function _View_FabButton0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_FabButton0, renderType_FabButton, import6.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckOnce);
    }
    _View_FabButton0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._el_0 = this.renderer.createElement(parentRenderNode, 'ion-icon', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'fab-close-icon');
        this.renderer.setElementAttribute(this._el_0, 'name', 'close');
        this.renderer.setElementAttribute(this._el_0, 'role', 'img');
        this._Icon_0_3 = new import12.Icon(this.parentInjector.get(import8.Config), new import9.ElementRef(this._el_0), this.renderer);
        this._el_1 = this.renderer.createElement(parentRenderNode, 'span', null);
        this.renderer.setElementAttribute(this._el_1, 'class', 'button-inner');
        this.renderer.projectNodes(this._el_1, import4.flattenNestedViewRenderNodes(this.projectableNodes[0]));
        this._el_2 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_2, 'class', 'button-effect');
        this._expr_0 = import7.UNINITIALIZED;
        this._expr_1 = import7.UNINITIALIZED;
        this.init([], [
            this._el_0,
            this._el_1,
            this._el_2
        ], [], []);
        return null;
    };
    _View_FabButton0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import12.Icon) && (0 === requestNodeIndex))) {
            return this._Icon_0_3;
        }
        return notFoundResult;
    };
    _View_FabButton0.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_0 = 'close';
        if (import4.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this._Icon_0_3.name = currVal_0;
            this._expr_0 = currVal_0;
        }
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_1 = this._Icon_0_3._hidden;
        if (import4.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this.renderer.setElementClass(this._el_0, 'hide', currVal_1);
            this._expr_1 = currVal_1;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_FabButton0.prototype.destroyInternal = function () {
        this._Icon_0_3.ngOnDestroy();
    };
    return _View_FabButton0;
}(import1.AppView));
export function viewFactory_FabButton0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_FabButton === null)) {
        (renderType_FabButton = viewUtils.createRenderComponentType('/home/nicolas/projects/cloudtripper/cloudtripper-frontend/node_modules/ionic-angular/components/fab/fab.d.ts class FabButton - inline template', 1, import10.ViewEncapsulation.None, styles_FabButton, {}));
    }
    return new _View_FabButton0(viewUtils, parentInjector, declarationEl);
}
var renderType_FabContainer_Host = null;
var _View_FabContainer_Host0 = (function (_super) {
    __extends(_View_FabContainer_Host0, _super);
    function _View_FabContainer_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_FabContainer_Host0, renderType_FabContainer_Host, import6.ViewType.HOST, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_FabContainer_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('ion-fab', rootSelector, null);
        this._appEl_0 = new import2.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_FabContainer0(this.viewUtils, this.injector(0), this._appEl_0);
        this._FabContainer_0_4 = new import3.FabContainer(new import9.ElementRef(this._el_0));
        this._query_FabButton_0_0 = new import13.QueryList();
        this._query_FabList_0_1 = new import13.QueryList();
        this._appEl_0.initComponent(this._FabContainer_0_4, [], compView_0);
        this._query_FabButton_0_0.reset([]);
        this._FabContainer_0_4._mainButton = this._query_FabButton_0_0.first;
        compView_0.create(this._FabContainer_0_4, this.projectableNodes, null);
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_FabContainer_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import3.FabContainer) && (0 === requestNodeIndex))) {
            return this._FabContainer_0_4;
        }
        return notFoundResult;
    };
    _View_FabContainer_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        this.detectContentChildrenChanges(throwOnChange);
        if (!throwOnChange) {
            if (this._query_FabList_0_1.dirty) {
                this._query_FabList_0_1.reset([]);
                this._FabContainer_0_4._fabLists = this._query_FabList_0_1;
                this._query_FabList_0_1.notifyOnChanges();
            }
            if ((this.numberOfChecks === 0)) {
                this._FabContainer_0_4.ngAfterContentInit();
            }
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_FabContainer_Host0.prototype.destroyInternal = function () {
        this._FabContainer_0_4.ngOnDestroy();
    };
    return _View_FabContainer_Host0;
}(import1.AppView));
function viewFactory_FabContainer_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_FabContainer_Host === null)) {
        (renderType_FabContainer_Host = viewUtils.createRenderComponentType('', 0, import10.ViewEncapsulation.None, [], {}));
    }
    return new _View_FabContainer_Host0(viewUtils, parentInjector, declarationEl);
}
export var FabContainerNgFactory = new import11.ComponentFactory('ion-fab', viewFactory_FabContainer_Host0, import3.FabContainer);
var styles_FabContainer = [];
var renderType_FabContainer = null;
var _View_FabContainer0 = (function (_super) {
    __extends(_View_FabContainer0, _super);
    function _View_FabContainer0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_FabContainer0, renderType_FabContainer, import6.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_FabContainer0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this.renderer.projectNodes(parentRenderNode, import4.flattenNestedViewRenderNodes(this.projectableNodes[0]));
        this.init([], [], [], []);
        return null;
    };
    return _View_FabContainer0;
}(import1.AppView));
export function viewFactory_FabContainer0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_FabContainer === null)) {
        (renderType_FabContainer = viewUtils.createRenderComponentType('/home/nicolas/projects/cloudtripper/cloudtripper-frontend/node_modules/ionic-angular/components/fab/fab.d.ts class FabContainer - inline template', 1, import10.ViewEncapsulation.None, styles_FabContainer, {}));
    }
    return new _View_FabContainer0(viewUtils, parentInjector, declarationEl);
}
