// ionic-angular/components/slides/slides.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/linker/element';
import * as import3 from 'ionic-angular/components/slides/slides';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from 'ionic-angular/config/config';
import * as import9 from '@angular/core/src/linker/element_ref';
import * as import10 from '@angular/core/src/metadata/view';
import * as import11 from '@angular/core/src/linker/component_factory';
var renderType_Slides_Host = null;
var _View_Slides_Host0 = (function (_super) {
    __extends(_View_Slides_Host0, _super);
    function _View_Slides_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Slides_Host0, renderType_Slides_Host, import6.ViewType.HOST, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_Slides_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('ion-slides', rootSelector, null);
        this._appEl_0 = new import2.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_Slides0(this.viewUtils, this.injector(0), this._appEl_0);
        this._Slides_0_4 = new import3.Slides(this.parentInjector.get(import8.Config), new import9.ElementRef(this._el_0), this.renderer);
        this._appEl_0.initComponent(this._Slides_0_4, [], compView_0);
        compView_0.create(this._Slides_0_4, this.projectableNodes, null);
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_Slides_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import3.Slides) && (0 === requestNodeIndex))) {
            return this._Slides_0_4;
        }
        return notFoundResult;
    };
    _View_Slides_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        if (((this.numberOfChecks === 0) && !throwOnChange)) {
            this._Slides_0_4.ngOnInit();
        }
        this.detectContentChildrenChanges(throwOnChange);
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_Slides_Host0;
}(import1.AppView));
function viewFactory_Slides_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Slides_Host === null)) {
        (renderType_Slides_Host = viewUtils.createRenderComponentType('', 0, import10.ViewEncapsulation.None, [], {}));
    }
    return new _View_Slides_Host0(viewUtils, parentInjector, declarationEl);
}
export var SlidesNgFactory = new import11.ComponentFactory('ion-slides', viewFactory_Slides_Host0, import3.Slides);
var styles_Slides = [];
var renderType_Slides = null;
var _View_Slides0 = (function (_super) {
    __extends(_View_Slides0, _super);
    function _View_Slides0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Slides0, renderType_Slides, import6.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckOnce);
    }
    _View_Slides0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._el_0 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'swiper-container');
        this._el_1 = this.renderer.createElement(this._el_0, 'div', null);
        this.renderer.setElementAttribute(this._el_1, 'class', 'swiper-wrapper');
        this.renderer.projectNodes(this._el_1, import4.flattenNestedViewRenderNodes(this.projectableNodes[0]));
        this._el_2 = this.renderer.createElement(this._el_0, 'div', null);
        this.renderer.setElementAttribute(this._el_2, 'class', 'swiper-pagination');
        this._expr_0 = import7.UNINITIALIZED;
        this.init([], [
            this._el_0,
            this._el_1,
            this._el_2
        ], [], []);
        return null;
    };
    _View_Slides0.prototype.detectChangesInternal = function (throwOnChange) {
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_0 = !this.context.showPager;
        if (import4.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this.renderer.setElementClass(this._el_2, 'hide', currVal_0);
            this._expr_0 = currVal_0;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_Slides0;
}(import1.AppView));
export function viewFactory_Slides0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Slides === null)) {
        (renderType_Slides = viewUtils.createRenderComponentType('/home/nicolas/projects/cloudtripper/cloudtripper-frontend/node_modules/ionic-angular/components/slides/slides.d.ts class Slides - inline template', 1, import10.ViewEncapsulation.None, styles_Slides, {}));
    }
    return new _View_Slides0(viewUtils, parentInjector, declarationEl);
}
var renderType_Slide_Host = null;
var _View_Slide_Host0 = (function (_super) {
    __extends(_View_Slide_Host0, _super);
    function _View_Slide_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Slide_Host0, renderType_Slide_Host, import6.ViewType.HOST, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_Slide_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('ion-slide', rootSelector, null);
        this._appEl_0 = new import2.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_Slide0(this.viewUtils, this.injector(0), this._appEl_0);
        this._Slide_0_4 = new import3.Slide(new import9.ElementRef(this._el_0), this.parentInjector.get(import3.Slides));
        this._appEl_0.initComponent(this._Slide_0_4, [], compView_0);
        compView_0.create(this._Slide_0_4, this.projectableNodes, null);
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_Slide_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import3.Slide) && (0 === requestNodeIndex))) {
            return this._Slide_0_4;
        }
        return notFoundResult;
    };
    _View_Slide_Host0.prototype.destroyInternal = function () {
        this._Slide_0_4.ngOnDestroy();
    };
    return _View_Slide_Host0;
}(import1.AppView));
function viewFactory_Slide_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Slide_Host === null)) {
        (renderType_Slide_Host = viewUtils.createRenderComponentType('', 0, import10.ViewEncapsulation.None, [], {}));
    }
    return new _View_Slide_Host0(viewUtils, parentInjector, declarationEl);
}
export var SlideNgFactory = new import11.ComponentFactory('ion-slide', viewFactory_Slide_Host0, import3.Slide);
var styles_Slide = [];
var renderType_Slide = null;
var _View_Slide0 = (function (_super) {
    __extends(_View_Slide0, _super);
    function _View_Slide0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Slide0, renderType_Slide, import6.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckOnce);
    }
    _View_Slide0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._el_0 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'slide-zoom');
        this.renderer.projectNodes(this._el_0, import4.flattenNestedViewRenderNodes(this.projectableNodes[0]));
        this.init([], [this._el_0], [], []);
        return null;
    };
    return _View_Slide0;
}(import1.AppView));
export function viewFactory_Slide0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Slide === null)) {
        (renderType_Slide = viewUtils.createRenderComponentType('/home/nicolas/projects/cloudtripper/cloudtripper-frontend/node_modules/ionic-angular/components/slides/slides.d.ts class Slide - inline template', 1, import10.ViewEncapsulation.None, styles_Slide, {}));
    }
    return new _View_Slide0(viewUtils, parentInjector, declarationEl);
}
